selfdrive/hardware/tici/agnos.py

```diff
@@ -154,21 +212,39 @@ def flash_agnos_update(manifest_path, cloudlog, spinner=None):
       cloudlog.info(f"Failed to flash {partition['name']}, aborting")
       raise Exception("Maximum retries exceeded")
 
-  cloudlog.info(f"AGNOS ready on slot {target_slot}")
+  cloudlog.info(f"AGNOS ready on slot {target_slot_number}")
+
+
+def verify_agnos_update(manifest_path, target_slot_number):
+  update = json.load(open(manifest_path))
+  return all(verify_partition(target_slot_number, partition) for partition in update)
 
 
 if __name__ == "__main__":
   import logging
   import time
-  import sys
+  import argparse
+
+  parser = argparse.ArgumentParser(description="Flash and verify AGNOS update",
+                                   formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
-  if len(sys.argv) != 2:
-    print("Usage: ./agnos.py <manifest.json>")
-    exit(1)
+  parser.add_argument("--swap", action="store_true", help="Verify and perform swap, downloads if necessary")
+  parser.add_argument("manifest", help="Manifest json")
+  args = parser.parse_args()
 
   spinner = Spinner()
   spinner.update("Updating AGNOS")
   time.sleep(5)
 
   logging.basicConfig(level=logging.INFO)
```

Comment:
should we try to import cloudlog?

Reply:
Think that relies on quite a few built stuff. I'm worried it might import successfully, but then fail in subtle ways.